src/breakupModel/simulation/Collision.cpp

```diff
@@ -39,17 +39,59 @@ void Collision::generateFragments() {
     }
 
     //The fragment Count, respectively Equation 4
-    auto fragmentCount = static_cast<size_t>(0.1 * std::pow(mass, 0.75) * std::pow(_minimalCharacteristicLength, -1.71));
-
-    const std::string debrisName{sat1.getName() + " & " + sat2.getName() + "-Collision-Fragment"};
-    this->createFragments(fragmentCount, debrisName, sat1.getPosition());
+    auto fragmentCount = static_cast<size_t>(0.1 * std::pow(mass, 0.75) *
+                                             std::pow(_minimalCharacteristicLength, -1.71));
+    this->createFragments(fragmentCount, sat1.getPosition());
 }
 
 void Collision::characteristicLengthDistribution() {
     //The pdf for Collisions is: 0.0101914/(x^2.71)
     Breakup::characteristicLengthDistribution(-2.71);
 }
 
+void Collision::assignParent() {
+    //The names of the fragments for a given parent
+    const Satellite &bigSat = _input.at(0);
+    const Satellite &smallSat = _input.at(1);
+    auto debrisNameBigPtr = std::make_shared<const std::string>(bigSat.getName() + "-Collision-Fragment");
+    auto debrisNameSmallPtr = std::make_shared<const std::string>(smallSat.getName() + "-Collision-Fragment");
+
+    //First specify the total mass
+    double inputMass = bigSat.getMass() + smallSat.getMass();
+    double resultMass = 0;
+    auto satIt = _output.begin();
+    for (; satIt != _output.end() && resultMass < inputMass; ++satIt) {
+        resultMass += satIt->getMass();
+    }
+    //Erase potential "too-much" mass
+    resultMass = resultMass > inputMass ? resultMass - (satIt - 1)->getMass() : resultMass;
+    _output.erase(satIt, _output.end());
+
+    double assignedBigMass = 0;
+    //All debris greater than the small satellite must origin from the big one
+    for (auto &sat : _output) {
+        if (sat.getCharacteristicLength() > smallSat.getCharacteristicLength()) {
+            sat.setName(debrisNameBigPtr);
+            sat.setVelocity(bigSat.getVelocity());
+            assignedBigMass += sat.getMass();
+        }
+    }
+
+    //Assign the rest with respect to the already assigned debris-mass for the big satellite
+    //first if: the mass of the bigSat is normed to the actual produced mass of the simulation
+    for (auto &sat : _output) {
```

Comment:
Is the debris sorted in some way? E.g. by characteristic length? Because then, e.g., this could severely bias the directional outcomes with all small debris flying in one and rest in the other direction if I understand correctly?

Reply:
Luckily, it's not sorted :)

Nevertheless, I add this in doc to draw attention to the fact that the _output does not have to be sorted when this method is called